Add automatic EPUB and PDF generation

f0lie opened the pull request:
The codebase was also cleaned up during this feature add.

PDF will be tackled next.

I didn't run the updated github action because I don't have github actions setup on my forked repo. The bash command works locally on Mac OS. I didn't verify if it would work on ubuntu, it seems like it should though.

f0lie committed:
Move markdown files into seperate folder.

f0lie committed:
Clean up function a little bit.

f0lie committed:
Fix deprecation warning.

f0lie committed:
Refactor some code so it's easier to change later.

f0lie committed:
Added meta data block to markdown for pandoc conversion later.

f0lie committed:
Refactored code.

f0lie committed:
Refactor metablock code.

f0lie committed:
Refactor original code for intented markdown files because the epub files don't actually need intentation.

The intention is still useful for just reading the markdown as it is.

f0lie committed:
Added EPUB generation using pandocs.

f0lie committed:
Added submission links to original reddit posts.

f0lie committed:
Merge remote-tracking branch 'upstream/main'

f0lie committed:
Updated README.md

f0lie committed:
Update README.md again

uruvelakassapa commented:
This looks great. Just one minor thing I noticed is that the markdown files are also there in the epub folder. If you can remove them I'll accept since I cannot do that myself because I'm away from my computer.

uruvelakassapa commented:
@f0lie actually maybe another thing to check would be if we have to `apt get install pandoc` or the like. You may be able to manually trigger the GitHub action on your forked repo to test with only a couple of clicks of config, if that.

f0lie committed:
Generate pdfs

f0lie committed:
Fixed broken github action because of double quotes within f-string

f0lie committed:
Fixed double quote issue.

f0lie committed:
Fix readme.md

f0lie committed:
Update github action to use latest python version.

f0lie committed:
Ignore .secrets file that store information for local github workflow runs.

f0lie committed:
Added pdf generation and updated the workflow to be up to date.

The .gitignore was updated to ignore files used to test locally working the workflow using act.

f0lie committed:
Updated git push command to be more verbose for debugging later.

f0lie committed:
Fixed problem where venv was being ignored by the python script because each run uses a different shell in github workflow.

f0lie committed:
Add sudo back into workflow.

f0lie committed:
Added doc about running github workflow locally.

f0lie committed:
Added sudo to the right places to fix workflow.

actions-user committed:
Update Reddit comments and markdown files

f0lie commented:
I have added PDF generation along with the EPUB generation. I used Source Serif 4 and used book layout of xelatex so the pdf's look quite nice and readable.

Removed the temp files.

Ran the github workflow to make sure it worked, which was quite an adventure actually.
